src/Scanner.js

```diff
@@ -430,12 +438,14 @@ class DefaultScanner {
         // find the enclosing parent form, and scan it.
         if (realTarget instanceof HTMLInputElement && !realTarget.hasAttribute(ATTR_INPUT_TYPE)) {
             const parentForm = this.getParentForm(realTarget);
-            if (parentForm && parentForm instanceof HTMLFormElement) {
-                const hasShadowTree = event.target?.shadowRoot != null;
-                const form = new Form(parentForm, realTarget, this.device, this.matching, this.shouldAutoprompt, hasShadowTree);
-                this.forms.set(parentForm, form);
-                this.findEligibleInputs(parentForm);
-            }
+
+            // If the parent form is an input element or the same as the target, we bail.
+            if (parentForm instanceof HTMLInputElement) return;
```

Comment:
The comment has two conditionals, but the code has only one. I think the code is correct, so you might want to update the comment. Also worth explaining why we bail?

Reply:
Yeah don't need to check both!